test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go

```diff
@@ -2,39 +2,37 @@ package profilesupdate
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"reflect"
 
+	. "github.com/onsi/gomega"
+
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 
 	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
 	profilecontroller "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/profile"
-	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
 	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
-	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
 )
 
 // UpdateIsolatedReservedCpus Updates the current performance profile with new sets of isolated and reserved cpus, and returns true if the update was successfull and false otherwise
-func UpdateIsolatedReservedCpus(isolatedSet performancev2.CPUSet, reservedSet performancev2.CPUSet) error {
-	profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
-	if err != nil {
-		return fmt.Errorf("could not get the performance profile: %v", err)
-	}
-	updatedProfile := profile.DeepCopy()
-	updatedProfile.Spec.CPU = &performancev2.CPU{
+func UpdateIsolatedReservedCpus(profile *performancev2.PerformanceProfile, isolatedSet performancev2.CPUSet, reservedSet performancev2.CPUSet) error {
+
+	profile.Spec.CPU = &performancev2.CPU{
```

Comment:
this seems a step back though. Why is trusting the given profile more reliable than getting a fresh copy?

Reply:
we are getting a fresh copy of the profile and passing that profile to this function.  We are doing this at:

https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go#L51